code/script.js

```diff
@@ -262,38 +295,53 @@ const selectQuestion = () => {
   // This variable stores what option group (category) the question belongs to.
   // We also need a variable that stores the actual value of the question we've selected.
 
+  const value = questions.value
   if (category === 'hair color') {
     currentQuestion = {
       attribute: 'hairColor',
-      // value: ,
+      value: value,
       // 👆 add the value from the input here
       category: category,
     }
   } else if (category === 'eye color') {
-    // Set this up your self
+    currentQuestion = {
+      attribute: 'eyeColor',
+      value: value,
+      category: category,
+    }
   } else if (category === 'accessories') {
     currentQuestion = {
-      //attribute: ,
+      attribute: value,
       // 👆 this is the property of the booleans such as smoke, glasses and hat. add the value from the input here
       value: true, // we're asking if this person wears a hat for exaple, so always true in the question.
       category: category,
     }
   } else if (category === 'other') {
+    currentQuestion = {
+      attribute: value,
+      value: true,
+      category: category,
+    }
     // Set this up your self (should be same structure as above)
   }
 }
 
+
+
 // This function should be invoked when you click on 'Find Out'.
 const checkQuestion = () => {
   // Compare the currentQuestion with the secret person.
+  const keep = currentQuestion.value === secret[currentQuestion.attribute];
```

Comment:
This is a nice compact way of establishing keep as a function rather than the if/else statements. Nicely done. :)